ovs/src/main/java/org/dcsa/conformance/standards/ovs/checks/OvsChecks.java

```diff
@@ -0,0 +1,300 @@
+        OvsFilterParameter.VESSEL_NAME,
+        "*/vesselSchedules/*/vesselName")
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate carrierVoyageNumber exists and " +
+        "matches in JSON response if request parameter has carrierVoyageNumber",
+      body -> validateParameter(
+        body,
+        sspSupplier,
+        OvsFilterParameter.CARRIER_VOYAGE_NUMBER,
+        "*/vesselSchedules/*/transportCalls/*/carrierExportVoyageNumber",
+        "*/vesselSchedules/*/transportCalls/*/carrierImportVoyageNumber")
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate universalVoyageReference exists and" +
+        " matches in JSON response if request parameter has universalVoyageReference",
+      body -> validateParameter(
+        body,
+        sspSupplier,
+        OvsFilterParameter.UNIVERSAL_VOYAGE_REFERENCE,
+        "*/vesselSchedules/*/transportCalls/*/universalImportVoyageReference",
+        "*/vesselSchedules/*/transportCalls/*/universalExportVoyageReference")
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate UNLocationCode exists and " +
+        "matches in JSON response if request parameter has UNLocationCode",
+      body -> validateParameter(
+        body,
+        sspSupplier,
+        OvsFilterParameter.UN_LOCATION_CODE,
+        "*/vesselSchedules/*/transportCalls/*/location/UNLocationCode")
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate facilitySMDGCode exists and " +
+        "matches in JSON response if request parameter has facilitySMDGCode",
+      body -> validateParameter(
+        body,
+        sspSupplier,
+        OvsFilterParameter.FACILITY_SMDG_CODE,
+        "*/vesselSchedules/*/transportCalls/*/location/facilitySMDGCode")
+    ));
+
+    checks.add(
+      JsonAttribute.customValidator(
+        "Validate startDate exists and " +
+          "matches in JSON response if request parameter has startDate",
+        body ->
+          validateDate(
+            body,
+            sspSupplier,
+            OvsFilterParameter.START_DATE,
+            (eventDate, expectedStartDate) ->
+              !eventDate.isAfter(expectedStartDate)))); // Check if eventDate is before or equal to startDate
+
+    checks.add(
+      JsonAttribute.customValidator(
+        "Validate endDate exists and " +
+          "matches in JSON response if request parameter has endDate",
+        body ->
+          validateDate(
+            body,
+            sspSupplier,
+            OvsFilterParameter.END_DATE,
+            (eventDate, expectedEndDate) ->
+              !eventDate.isBefore(expectedEndDate)))); // Check if eventDate is after or equal to endDate
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate transportCallReference is unique across each array node",
+      OvsChecks::validateUniqueTransportCallReference
+    ));
+
+    checks.add(JsonAttribute.customValidator(
+      "Validate limit parameter is met",
```

Comment:
I have no clue what this sentence means for the end user. Can you rephrase it, to make it clear? E.g.: limit parameter is wrong? Limit parameter is higher than allowed? I don't know.

Reply:
Ok.. will elaborate it more.